frontends/concrete-python/concrete/fhe/compilation/module_compiler.py

```diff
@@ -402,7 +469,7 @@ class Wired(NamedTuple):
     Composition policy which allows the forwarding of certain outputs to certain inputs.
     """
 
-    wires: List[Wire]
+    wires: Set[Wire] = set()
```

Comment:
Due to an interplay between the way the tracing is implemented, and the implicit inputset generation, the same rule may be added multiple times. This is no big deal, but it makes testing more difficult, and I thought it would trip up users if they saw that.

Reply:
that's what I understood, make sense :+1: